client/transaction-pool/src/testing/pool.rs

```diff
@@ -213,12 +220,80 @@ fn should_not_retain_invalid_hashes_from_retracted() {
 	pool.api.add_invalid(&xt);
 
 	block_on(pool.maintain(&BlockId::number(1), &[retracted_hash]));
+
+	// maintaince is in background
+	block_on(futures_timer::Delay::new(std::time::Duration::from_millis(10)));
```

Comment:
Can't we just call something directly to trigger it instead?

Reply:
This will require bunch of confusing code that will exists only for 1 line in 2 tests. I would prefer not to.

Reply:
There should be no issues with this, really

It is not like we wait for something that _approximately_ finishes in 10ms. We know for sure it will be done in 5ms, so no flakiness here.